Gigs/Gigs/ModelControllers/GigController.swift

```diff
@@ -0,0 +1,178 @@
+//
+//  GigController.swift
+//  Gigs
+//
+//  Created by Cora Jacobson on 7/11/20.
+//  Copyright © 2020 Cora Jacobson. All rights reserved.
+//
+
+import Foundation
+
+class GigController {
+    
+    enum HTTPMethod: String {
+        case get = "GET"
+        case post = "POST"
+    }
+    
+    enum NetworkError: Error {
+        case noData
+        case failedSignUp
+        case failedSignIn
+        case noToken
+        case tryAgain
+    }
+    
+    var bearer: Bearer?
+    
+    var gigs: [Gig] = []
+    
+    private let baseURL = URL(string: "https://lambdagigs.vapor.cloud/api")!
+    private lazy var signUpURL = baseURL.appendingPathComponent("/users/signup")
+    private lazy var signInURL = baseURL.appendingPathComponent("/users/login")
+    private lazy var gigsURL = baseURL.appendingPathComponent("/gigs/")
+    
+    func signUp(with user: User, completion: @escaping (Result<Bool, NetworkError>) -> Void) {
+        print("signUpURL = \(signUpURL.absoluteString)")
+        var request = postRequest(for: signUpURL)
+        do {
+            let jsonData = try JSONEncoder().encode(user)
+            print(String(data: jsonData, encoding: .utf8)!)
+            request.httpBody = jsonData
+            let task = URLSession.shared.dataTask(with: request) { (_, response, error) in
+                if let error = error {
+                    print("SignUp failed with error: \(error)")
+                    completion(.failure(.failedSignUp))
+                    return
+                }
+                guard let response = response as? HTTPURLResponse,
+                    response.statusCode == 200 else {
+                        print("Sign up was unsuccesful")
+                        completion(.failure(.failedSignUp))
+                        return
+                }
+                completion(.success(true))
+            }
+            task.resume()
+        } catch {
+            print("Error encoding user: \(error)")
+            completion(.failure(.failedSignUp))
+        }
```

Comment:
I would prefer to only wrap the do catch block around the try statement so it is a little more clear what is going to throw an error. Nothing inherently wrong with this code, just a matter of style.